### ISSUE TITLE
Include git-lfs information in contributor setup documentation

### ISSUE THREAD

**Issue by lisavogtsf** [desktop/desktop#5415]
In attempting to push a branch to contribute I received the following error:
![image](https://user-images.githubusercontent.com/7121497/44176157-8b2b9900-a09d-11e8-8f31-5b5205b07dd5.png)
```
This repository is configured for Git LFS but 'git-lfs' was not found on your path. If you no longer wish to use Git LFS, remove this hook by deleting .git/hooks/post-checkout.
```

This was easy enough to resolve by googling for and installing git-lfs via https://git-lfs.github.com/, but there is no mention of git-lfs in the contributor setup documentation, and I think there should be.

I suggest adding a reference to git-lfs to the [setup.md](https://github.com/desktop/desktop/blob/master/docs/contributing/setup.md) file. It may not be as important as getting python in, but at least mentioning that you need to have it installed would be nice.

**Comment by shiftkey** [comment]
@lisavogtsf thanks for the heads up. We don't have any rules set for Git LFS in our `.gitattributes` file, so I think we can live without this. A `git lfs uninstall --local` should cleanup the hooks in your directory so that error goes away.

I'll check the repository setup and confirm we don't need this, and see if we can figure out where it was setup. Did you clone the repository in Desktop too?

**Comment by lisavogtsf** [comment]
@shiftkey I'm not sure what you're asking: "Did you clone the repository in Desktop too?"

I expect that I cloned the repository using the CLI: 
```
git clone https://github.com/desktop/desktop.git github-desktop
```
I did not separately clone the repository using the Github Desktop client. Does that answer the question?

Knowing very little about git lfs, I wonder how my repo got configured to use it. The .git/hooks/pre-push file in the error has no mention of git lfs. I'm a bit perplexed.

(I like the 'meta' tag!)

**Comment by shiftkey** [comment]
@lisavogtsf 

> I did not separately clone the repository using the Github Desktop client. Does that answer the question?

Yep, that's fine. I'm just curious how those hooks were added to your `.git` directory, and maybe Desktop was eagerly (and incorrectly) doing it. I checked one of my local Desktop clones and it looked like `git lfs install --local` had been run:

```
$ git lfs install --local
Updated git hooks.
Git LFS initialized.

$ ls .git/hooks
applypatch-msg.sample     post-update.sample        pre-rebase.sample
commit-msg.sample         pre-applypatch.sample     pre-receive.sample
post-checkout             pre-commit.sample         prepare-commit-msg.sample
post-commit               pre-push                  update.sample
post-merge                pre-push.sample

$ cat .git/hooks/pre-push
#!/bin/sh
command -v git-lfs >/dev/null 2>&1 || { echo >&2 "\nThis repository is configured for Git LFS but 'git-lfs' was not found on your path. If you no longer wish to use Git LFS, remove this hook by deleting .git/hooks/pre-push.\n"; exit 2; }
git lfs pre-push "$@"
```

> The .git/hooks/pre-push file in the error has no mention of git lfs and neither does .gitattributes. I'm a bit perplexed.

What's in there? Maybe I can help diagnose where it came from?

**Comment by lisavogtsf** [comment]
I'm not sure how much more time this is worth, but I'm having a slow afternoon. 

To clarify, I did not initially have git-lfs installed, and that was when I was getting the error trying to push back to my fork of the desktop/desktop repo.

I do now have git-lfs:
```
$ git-lfs -v
git-lfs/2.5.1 (GitHub; darwin amd64; go 1.10.3)

$ which git-lfs
/usr/local/bin/git-lfs
```

Here's .gitattributes:
```
$ cat .gitattributes
/app/test/fixtures/** -text
```

Ok, for me it looks like this, with no `pre-push` file only a `pre-push.sample`:

```
$ ls .git/hooks
applypatch-msg.sample     post-update.sample        pre-push.sample           prepare-commit-msg.sample
commit-msg.sample         pre-applypatch.sample     pre-rebase.sample         update.sample
fsmonitor-watchman.sample pre-commit.sample         pre-receive.sample
```
The contents of that sample file are somewhat long and I'm not sure how useful they are: 
```
$ cat .git/hooks/pre-push.sample
#!/bin/sh

# An example hook script to verify what is about to be pushed.  Called by "git
# push" after it has checked the remote status, but before anything has been
# pushed.  If this script exits with a non-zero status nothing will be pushed.
#
# This hook is called with the following parameters:
#
# $1 -- Name of the remote to which the push is being done
# $2 -- URL to which the push is being done
#
# If pushing without using a named remote those arguments will be equal.
#
# Information about the commits which are being pushed is supplied as lines to
# the standard input in the form:
#
#   <local ref> <local sha1> <remote ref> <remote sha1>
#
# This sample shows how to prevent push of commits where the log message starts
# with "WIP" (work in progress).

remote="$1"
url="$2"

z40=0000000000000000000000000000000000000000

while read local_ref local_sha remote_ref remote_sha
do
	if [ "$local_sha" = $z40 ]
	then
		# Handle delete
		:
	else
		if [ "$remote_sha" = $z40 ]
		then
			# New branch, examine all commits
			range="$local_sha"
		else
			# Update to existing branch, examine new commits
			range="$remote_sha..$local_sha"
		fi

		# Check for WIP commit
		commit=`git rev-list -n 1 --grep '^WIP' "$range"`
		if [ -n "$commit" ]
		then
			echo >&2 "Found WIP commit in $local_ref, not pushing"
			exit 1
		fi
	fi
done

exit 0
```

**Comment by shiftkey** [comment]
@lisavogtsf Git ignores the `.sample` files when a repository is initialized. They're there to have a starting point for writing hooks.

I'm happy to close this out if there's nothing else to troubleshoot from, but if someone else stumbles on a Desktop repo that's somehow initialized for Git LFS I'd love to get to the bottom of it because we don't need it. @lisavogtsf anything else I've missed from this?

**Comment by lisavogtsf** [comment]
@shiftkey I'm pretty unfamiliar with how git-lfs and git hooks work. I think we can close this, but I'm pretty perplexed about where the hooks on my machine came from. 

As a final experiment, when I ran `brew uninstall git-lfs` and confirmed it was gone with `$ git-lfs -v` I tried updating my local repo from desktop/desktop then tried pushing that back up to my fork... and it worked fine with no git lfs error. So I guess something to keep in mind if other's have this trouble during first-time setup.

Thanks!
